commit e9d1af168a907f7cb0c80904c73965d24fc7e9b9 (refs/changes/39/5639/1)
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-02-26T14:33:52+00:00 (7 months ago)
    
    TOOLS-2188 bits-upload.sh should pass --compression=none for uncompressed image files

diff --git a/tools/bits-upload.sh b/tools/bits-upload.sh
index 86f46d2..a42bf94 100755
--- a/tools/bits-upload.sh
+++ b/tools/bits-upload.sh
@@ -290,6 +290,7 @@ function find_upload_bits {
 function publish_to_updates {
     local manta_path
     local msigned_url
+    local compression_flag
 
     echo "Publishing updates to updates.joyent.com"
     for file in ${FILES}; do
@@ -303,6 +304,7 @@ function publish_to_updates {
 
         MF=${file}
         IMAGEFILE=$(echo ${MF} | sed -e 's/\..*manifest$/.zfs.gz/g')
+        compression_flag=""
 
         # Some payloads are not zfs-based, look for likely alternatives.
         # This assumes that a single directory with <file>.manifest
@@ -310,6 +312,7 @@ function publish_to_updates {
 	# <file>.tar.gz]
         if [[ ! -f "${IMAGEFILE}" ]]; then
             IMAGEFILE=$(echo ${MF} | sed -e 's/\..*manifest$/.sh/g')
+	    compression_flag="--compression=none"
         fi
         if [[ ! -f "${IMAGEFILE}" ]]; then
             IMAGEFILE=$(echo ${MF} | sed -e 's/\..*manifest$/.tgz/g')
@@ -367,8 +370,8 @@ function publish_to_updates {
         echo "UPDATES_IMGADM_CHANNEL=$UPDATES_IMGADM_CHANNEL"
         echo "UPDATES_IMGADM_USER=$UPDATES_IMGADM_USER"
         echo "UPDATES_IMGADM_IDENTITY=$UPDATES_IMGADM_IDENTITY"
-        echo Running: ${UPDATES_IMGADM} import -m ${MF} -f "${msigned_url}"
-        ${UPDATES_IMGADM} import -m ${MF} -f "${msigned_url}"
+        echo Running: ${UPDATES_IMGADM} import -m ${MF} -f "${msigned_url}" ${compression_flag}
+        ${UPDATES_IMGADM} import -m ${MF} -f "${msigned_url}" ${compression_flag}
     done
 }
 
